File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DiagnosticsComparer.Primary.vb
Public Class TabCharacter
    Public Sub CorrectNoncompliant()
        Dim foo = New String() {"a", "b", "c"} ' Noncompliant {{Use an array literal here instead.}}
'                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    End Sub

    Public Sub Unexpected()
        Dim foo = New String() {"a", "b", "c"}
    End Sub

    Public Sub WrongMessage()
        Dim foo = New String() {"a", "b", "c"} ' Noncompliant {{Wrong message}}
    End Sub

    Public Sub WrongStart()
        Dim foo = New String() {"a", "b", "c"} ' Noncompliant
'                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    End Sub

    Public Sub WrongLength()
        Dim foo = New String() {"a", "b", "c"} ' Noncompliant {{Use an array literal here instead.}}
'                     ^^^^^^^^^^^^^^^^^^^^^^^^^
    End Sub
End Class

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DiagnosticsComparer.Primary.cs
public class FirstClass // Noncompliant
{
    async void CorrectNoncompliant() { } // Noncompliant {{Return 'Task' instead.}}
    //    ^^^^

    async void Unexpected() { }

    async void WrongMessage() { } // Noncompliant {{Wrong message}}

    async void WrongPlaceWrongMessage() { } // Noncompliant {{Wrong message, wrong place}}
    //         ^^^^^^^^^^

    async void WrongPlace() { } // Noncompliant {{Return 'Task' instead.}}
    //         ^^^^^^^^^^

    async void WrongPlaceNoMessage() { } // Noncompliant
    //         ^^^^^^^^^^

    async void WrongLength() { } // Noncompliant {{Return 'Task' instead.}}
    //    ^^^^^
}
